daemon/graphdriver/windows/windows.go

```diff
@@ -331,7 +332,23 @@ func (d *Driver) Remove(id string) error {
 	tmpID := fmt.Sprintf("%s-removing", rID)
 	tmpLayerPath := filepath.Join(d.info.HomeDir, tmpID)
 	if err := os.Rename(layerPath, tmpLayerPath); err != nil && !os.IsNotExist(err) {
-		return err
+		// If permission denied, it's possible that the scratch is still mounted, an
+		// artifact after a hard daemon crash for example. Worth a shot to try detaching it
+		// before retrying the rename.
+		if os.IsPermission(err) {
+			f := filepath.Join(layerPath, "sandbox.vhdx")
+			if _, exists := os.Stat(f); exists == nil {
```

Comment:
Kind of strange the call the error `exists`, since the true case is `nil`, but okay, I guess.

Reply:
Agree this is confusing to read. Does can we just call `DetachVhd` directly instead of checking existence first?

Reply:
I'll update to address both of these.